src/network/networkd-dhcp4.c

```diff
@@ -1071,6 +1071,12 @@ static int dhcp4_handler(sd_dhcp_client *client, int event, void *userdata) {
                         if (link->ipv4ll) {
                                 log_link_debug(link, "DHCP client is stopped. Acquiring IPv4 link-local address");
 
+                                if (link->network->ipv4ll_start_address.s_addr != 0) {
```

Comment:
This condition is not necessary, as `sd_ipv4ll_set_address()` handles that gracefully.